libraries/persistent_store/tests/store.rs

```diff
@@ -39,13 +41,81 @@ fn interrupted_overflowing_compaction() {
     // We trigger 2 interrupted overflowing compactions, which would move the last non-deleted entry
     // out of the window unless additional compactions are done to restore the overflow.
     for _ in 0..2 {
-        let interruption = StoreInterruption {
-            delay: 0,
-            corrupt: Box::new(|old, new| old.copy_from_slice(new)),
-        };
-        match driver.partial_apply(StoreOperation::Prepare { length: 1 }, interruption) {
+        match driver.partial_apply(
+            StoreOperation::Prepare { length: 1 },
+            StoreInterruption::pure(1),
+        ) {
+            Ok((None, d)) => driver = d.power_on().unwrap(),
+            _ => unreachable!(),
```

Comment:
Yes this is only in the test. I used `assert!(false)` initially but the compiler doesn't realize that the control flow always stops as `assert!(false)` so it complains that `driver` is not defined when taking that branch. But I guess a solution would be to have `{ assert!(false); return; }`. What do you think?

Reply:
Interesting. I like that a bit more. Reading the code, I was wondering: What is this even testing? At least me personally, I parse for `assert` variants.

Reply:
Ah I see. So no, this test is not testing that we don't reach this statement. It's testing that there's no panic which are in many places (including outside the test). The most important ones are those in the loop, i.e. the unwrap on the line above (which was the one failing before the bug) and the assert false on this line (which encompasses 2 failure types, unwrapping the partial apply and making sure that when it's Ok there's still no error).

Reply:
So the `assert` is still more precise than the `unreachable` right? I'm happy as is, and I don't think we have to document which ones are the "important" unwraps. Therefore resolving.

Reply:
It depends if we associate an additional semantics to assert than unreachable. They should be the same from compiler point of view, but it's true that in tests we are more used to see assert, so it might be less surprising.